src/gitingest/clone.py

```diff
@@ -103,42 +100,65 @@ async def clone_repo(config: CloneConfig) -> tuple[bytes, bytes]:
 
     try:
-        if commit:
-            # Scenario 1: Clone and checkout a specific commit
-            # Clone the repository without depth to ensure full history for checkout
-            clone_cmd = ["git", "clone", "--single-branch", url, local_path]
-            await run_git_command(*clone_cmd)
-
-            # Checkout the specific commit
-            checkout_cmd = ["git", "-C", local_path, "checkout", commit]
-            return await run_git_command(*checkout_cmd)
-
-        if branch and branch.lower() not in ("main", "master"):
-
-            # Scenario 2: Clone a specific branch with shallow depth
-            clone_cmd = ["git", "clone", "--depth=1", "--single-branch", "--branch", branch, url, local_path]
-            return await run_git_command(*clone_cmd)
-
-        # Scenario 3: Clone the default branch with shallow depth
-        clone_cmd = ["git", "clone", "--depth=1", "--single-branch", url, local_path]
-        return await run_git_command(*clone_cmd)
-
-    except (RuntimeError, asyncio.TimeoutError, AsyncTimeoutError):
-        raise  # Re-raise the exception
+        if query["commit"]:
+            proc = await asyncio.create_subprocess_exec(
+                "git",
+                "clone",
+                "--single-branch",
+                query["url"],
+                query["local_path"],
+                stdout=asyncio.subprocess.PIPE,
+                stderr=asyncio.subprocess.PIPE,
+            )
+            stdout, stderr = await proc.communicate()
+
+            proc = await asyncio.create_subprocess_exec(
+                "git",
+                "-C",
+                query["local_path"],
+                "checkout",
+                query["branch"],
+                stdout=asyncio.subprocess.PIPE,
+                stderr=asyncio.subprocess.PIPE,
+            )
+            stdout, stderr = await proc.communicate()
+        elif (
+            query["branch"] != "main"
+            and query["branch"] != "master"
+            and query["branch"]
+        ):
+            proc = await asyncio.create_subprocess_exec(
+                "git",
+                "clone",
+                "--depth=1",
+                "--single-branch",
+                "--branch",
+                query["branch"],
+                query["url"],
+                query["local_path"],
+                stdout=asyncio.subprocess.PIPE,
+                stderr=asyncio.subprocess.PIPE,
+            )
+        else:
+            proc = await asyncio.create_subprocess_exec(
+                "git",
+                "clone",
+                "--depth=1",
+                "--single-branch",
+                query["url"],
+                query["local_path"],
+                stdout=asyncio.subprocess.PIPE,
+                stderr=asyncio.subprocess.PIPE,
+            )
+
+        stdout, stderr = await proc.communicate()
+        return stdout, stderr
+    except TimeoutError:
```

Comment:
We kind of need to raise the error for the current web version to work, but I now realise that this is not ideal for the CLI